daffodil-japi/src/main/scala/org/apache/daffodil/japi/packageprivate/Utils.scala

```diff
@@ -73,6 +73,8 @@ private[japi] object ValidationConversions {
       case ValidationMode.Off => SValidationMode.Off
       case ValidationMode.Limited => SValidationMode.Limited
       case ValidationMode.Full => SValidationMode.Full
+      // todo;; support custom in java?
+      case _ => SValidationMode.Off
```

Comment:
Not sure how to approach this

Reply:
Suggest for JAPI, setValidationMode/withValidationMode takes an enum, and two other optional arguments. The enum gets an additional Custom, and when Custom is specified, the other two args, which are the name and list of args per your Custom(name., args) Scala ValidationMode.Custom case class constructor. 

These same additional objects get passed to modeToScala, which can actually then construct a Custom object like the CLI code does.

Reply:
Do we really need the Arguments for the API? Could the API be something maybe a bit simpler, e.g.

```scala
val myValidator = new MyCustomValidator(...)
dp.withValidation(Validation.Custom, myValidator)
```

So if you are using the API, then we expect you to allocate a new validator and we provide an API method to provide it to Daffodil. One potential issue is what if the validator isn't threadsafe (I believe Xerces isn't?), but perhaps the CustomValidator should be expected to handle that if they arent thread safe?

Reply:
SAPI has examples now, JAPI is WIP

Reply:
JAPI has examples.  Minor tweaks to the scala impl allowed sane usage.